airflow/providers/google/cloud/transfers/bigquery_to_mssql.py

```diff
@@ -0,0 +1,167 @@
+    :type dataset_table: str
+    :param selected_fields: List of fields to return (comma-separated). If
+        unspecified, all fields are returned.
+    :type selected_fields: str
+    :param gcp_conn_id: reference to a specific Google Cloud hook.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param mssql_conn_id: reference to a specific mssql hook
+    :type mssql_conn_id: str
+    :param database: name of database which overwrite defined one in connection
+    :type database: str
+    :param replace: Whether to replace instead of insert
+    :type replace: bool
+    :param batch_size: The number of rows to take in each batch
+    :type batch_size: int
+    :param location: The location used for the operation.
+    :type location: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = (
+        'dataset_id',
+        'table_id',
+        'mssql_table',
+        'impersonation_chain',
+    )
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,  # pylint: disable=too-many-arguments
+        dataset_table: str,
+        mssql_table: str,
+        selected_fields: Optional[str] = None,
+        gcp_conn_id: str = 'google_cloud_default',
+        mssql_conn_id: str = 'mssql_default',
+        database: Optional[str] = None,
+        delegate_to: Optional[str] = None,
+        replace: bool = False,
+        batch_size: int = 1000,
+        location: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.selected_fields = selected_fields
+        self.gcp_conn_id = gcp_conn_id
+        self.mssql_conn_id = mssql_conn_id
+        self.database = database
+        self.mssql_table = mssql_table
+        self.replace = replace
+        self.delegate_to = delegate_to
+        self.batch_size = batch_size
+        self.location = location
+        self.impersonation_chain = impersonation_chain
+        try:
+            self.dataset_id, self.table_id = dataset_table.split('.')
```

Comment:
How about `project.dataset.table` ? You can utilise method from bq library to parse string into database/table reference

Reply:
Thanks, let me check that.

Reply:
Im looking at https://googleapis.dev/python/bigquery/latest/reference.html and I can find the method to parse string into database/table, can u please point me to the right library.

Reply:
https://github.com/apache/airflow/blob/13faa6912f7cd927737a1dc15630d3bbaf2f5d4d/airflow/providers/google/cloud/hooks/bigquery.py#L690

Reply:
@turbaszek made the change.